.github/workflows/main.yml

```diff
@@ -16,34 +16,45 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: "3.8"
-      - uses: eifinger/setup-rye@v2.6
-      - run: rye config --set-bool behavior.use-uv=true
-      - run: rye sync --no-lock
+          version: "0.4.12"
+          enable-cache: true
+
+      - name: Set up Python 3.12
+        run: uv python install 3.12
+
+      - name: Install dependencies
+        run: uv sync --python 3.12 --frozen
```

Comment:
feels like linting should also be using latest dependencies

Reply:
I'm maintaining the same behavior here.

Reply:
OK, it doesn't have to change here, but I'm questioning it.

Reply:
I don't mind those, just the tests.